### PR TITLE
How to Pick PDF Files and Images from Phone Storage in Android Using Kotlin

### LINKED ISSUE #4137: [Android] How to Pick PDF Files and Images from Phone Storage in Android Using Kotlin
#### Proposed title of article
[Android] How to Pick PDF Files and Images from Phone Storage in Android Using Kotlin

#### Proposed article introduction
This tutorial is based on Android and is  going to help android developers to pick PDF files and images from the phone storage and display in the TextView and ImageView respectively. The tutorial is going to be in Kotlin programming language and also easy for beginners to read and understand.

Because the user is involved in selecting the files or directories that your app can access, this mechanism doesn't require any system permissions, and user control and privacy is enhanced. Additionally, these files, which are stored outside of an app-specific directory and outside of the media store, remain on the device after your app is uninstalled.

#### Key takeaways 
- Understanding explicit intent for navigating to the phone storage.
- How to choose images and PDF files.
- How to set the files in the Views.

#### Article quality
My article is unique since the content is not covered in the official documentation. It is also beginner friendly and written in Kotlin.

#### References
- https://developer.android.com/training/data-storage/shared/documents-files
- https://developer.android.com/guide/components/intents-filters

### PR DESCRIPTION
**Pull request by Hepatrique**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4137

### CODE REVIEW

**Comment by okeloviolet** [comment]
@Hepatrique  @hectorkambow I'll be reviewing this article.

**Comment by Hepatrique** [comment]
Hello @okeloviolet , @ahmadmardeni1 and @hectorkambow the article is ready for review. Please assign a reviewer as suggested by @okeloviolet . Thanks in advance.

**Comment by LinusMuema** [comment]
plagiarism check done :heavy_check_mark:

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
+This tutorial is going to cover on how to open gallery and files apps from your android application and pick an image and a PDF file. You will learn about intents specifically implicit intents.
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:1]: Change to "This tutorial will cover how to open gallery and files apps from your android application and pick an image and a PDF file. In addition, you will learn about intents, specifically implicit intents. "

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+### Goals
+- Picking images from gallery and loading to `ImageView`.
+- Taking photos and loading to `ImageView`.
+- Picking PDF file and display in a `TextView`. 
+
+File choosing in Android involves implicit intents. Implicit intent is a type of intent that navigates the user to another application. For File choosing to be successful, certain permissions MUST  be allowed in the app manifest.
+
+### What is Intents
+Is an object that facilitates communication between app components. Intents are used in cases when an application wants to start an activity, start a service, or when delivering a broadcast.
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:29]: Change to " Intent is an object that facilitates communication between app components. For example, intents are used when an application starts an activity, starts a service, or delivers a broadcast."

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+### Types of Intents
+Intents consist of two types:
+
+#### Implicit intents
+Is a type of intent that facilitates communication between two different apps. For instance, in this tutorial, we are going to imply implicit intent to communicate with the gallery and files apps. Implicit intents only need the declaration of the general action to be performed.
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:35]: Change to " Implicit intent is a type of intent that facilitates communication between two different apps. For instance, in this tutorial, we will imply implicit intent to communicate with the gallery and files apps. Implicit intents only need the declaration of the general action to be performed."

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+#### Explicit intents
+Is a type of intent that allows communication within an application itself. For example, when you want to navigate from one activity to the next, you will trigger an explicit intent. 
+
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:39]: Change to "
An explicit intent is a type of intent that allows communication between application components. For example, when you want to navigate one activity to the next, you will trigger an explicit intent. "

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+            }
+        })
+        myAlertDialog.show()
+    }
+```
+Below is a screenshot of the alert dialog.
+
+[Alert Dialog](/engineering-education/picking-pdf-and-image-from-phone-storage/alert-dialog.png)
+
+When you want only the images in your gallery to be displayed, you include the type of intent to be images but if you want to display videos as well do not specify the type.
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:126]: Change to " When you want only the images in your gallery to be displayed, you include the intent to be images, but if you want to display videos, do not specify the type."

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+                    }
+                }
+            }
+        }
+    }
+```
+[Screenshot](!/engineering-education/picking-pdf-and-image-from-phone-storage/screenshot.png)
+
+### Coclusion
+This tutorial is only meant to guide you through. You need to try the codes provided to practice and master the concept.
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:320]: Change to " This tutorial is only meant to guide you through. First, you need to try the codes provided to practice and master the concept. "

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+            }
+        }
+    }
+```
+[Screenshot](!/engineering-education/picking-pdf-and-image-from-phone-storage/screenshot.png)
+
+### Coclusion
+This tutorial is only meant to guide you through. You need to try the codes provided to practice and master the concept. 
+
+Happy codding!!!.
```
Comment (okeloviolet) [inline content/articles/picking-pdf-and-image-from-phone-storage/index.md:322]: Change to "Happy coding!"

**Review by okeloviolet — CHANGES_REQUESTED** [review]
@Hepatrique amazing article, fix the suggested changes and add link to complete source code.

File: content/articles/picking-pdf-and-image-from-phone-storage/index.md (hunk 7 above)
Comment (Hepatrique) [reply content/articles/picking-pdf-and-image-from-phone-storage/index.md]: Thanks @okeloviolet for the suggested corrections. Once done I will ping you

**Comment by Hepatrique** [comment]
Am done with  the changes

**Comment by hectorkambow** [comment]
@Hepatrique 
Please be sure you are going over the entire article for any other mistakes or errors, as our peer reviewers are not responsible to point out every mistake - that is the authors responsibility to properly review and proofread their work. 

This helps prevent much back and forth. 

cc @okeloviolet

**Comment by okeloviolet** [comment]
@Hepatrique please add inline comments to your code sections to explain various parts of your code, rename avartar.jpeg to avatar.jpeg

**Comment by Hepatrique** [comment]
> Once done the article will be ready for final review

Done with the changes

**Comment by hectorkambow** [comment]
will be jumping on this one.

**Comment by hectorkambow** [comment]
@Hepatrique please be sure to go over the edits made by your peer reviewer and myself. 

Please do the same to improve potential future reviews. @okeloviolet

**Pull request merged**